packages/manager/src/utilities/omittedProps.ts

```diff
@@ -13,3 +13,25 @@
 export const omittedProps = <Props>(props: Array<keyof Props>) => (
   prop: keyof Props
 ): boolean => !props.includes(prop);
+
+/**
+ * Helper to filter out props we spread into a component.
+ * This helpers differs from `omittedProps` in that it omits the props
+ * from the object instead of returning a boolean.
+ * This util is a direct replacement for `omit` from lodash.
+
+ * @param props Array of props to filter out
+ * @param toRemove Array of props to remove
+ * @returns Object with props removed
+ */
+export function omitProps<Props extends {}, Keys extends keyof Props>(
+  props: Props,
+  toRemove: Keys[]
+): Omit<Props, Keys> {
+  return Object.keys(props).reduce((obj, key) => {
+    if (!toRemove.includes(key as any)) {
+      (obj as any)[key] = props[key as keyof Props];
+    }
+    return obj;
+  }, {} as Omit<Props, Keys>);
+}
```

Comment:
Another way to implement with less casting:
```suggestion
export const omitProps = <Props extends {}, Keys extends keyof Props & string>(
  props: Props,
  toRemove: Keys[] & string[]
) =>
  Object.fromEntries(
    Object.entries(props).filter(([key]) => !toRemove.includes(key))
  ) as Omit<Props, Keys>;
```

Reply:
nice! cleaner indeed. changed